optuna/study/study.py

```diff
@@ -42,7 +43,7 @@
 
 _dataframe = _LazyImport("optuna.study._dataframe")
 
-if TYPE_CHECKING:
+if TYPE_CHECKING or "sphinx" in sys.modules:
     from optuna.study._dataframe import pd
```

Comment:
I'm not fully sure, but if users have Sphinx installed without Pandas, they might encounter an import error. Could you please look into this?

Reply:
When I uninstalled `pandas` and then performed `made html`, I got a warning from `study.py`, but `plotly` gave import errors regardless of this PR.
However, 
```
>>> import optuna
>>> import sys
>>> 'sphinx' in sys.modules
False
>>> import sphinx
>>> 'sphinx' in sys.modules
True
```
It seems to be not a good behavior.

Reply:
On second thought, I thought it would not be a problem. I think lazy-import will prevent import error.

Reply:
Thank you for your investigation.
BTW, this PR seems to be still work-in-progress, so let me remove the assignment for now.